src/pages/api/recipients/[recId]/supply-requests.ts

```diff
@@ -79,8 +89,62 @@ async function handler(
       } catch (e) {
         return serverErrorHandler(e, res);
       }
+    case 'POST':
+      try {
+        const prisma = new PrismaClient();
+        const recipientAddSupply = await prisma.recipient.findUnique({
+          where: {
+            id: Number(recId),
+          },
+        });
+
+        if (!recipientAddSupply) {
+          return res.status(400).json({
+            error: true,
+            message: `No recipient found for id: ${recId}`,
+          });
+        }
+
+        if (!isValidRecipient) {
```

Comment:
Nit: This should be the first thing you check in the endpoint (can save executing a query). Also, the status code here should be `401` for unauthorized users